extensions/wikia/UserRenameTool/process/ProcessGlobal.class.php

```diff
@@ -0,0 +1,597 @@
+				'errorMessage' => $e->getMessage(),
+				'errorLocation' => $e->getFile(),
+				'errorLine' => $e->getLine()
+			] );
+			$this->addError(
+				wfMessage( 'userrenametool-error-cannot-rename-unexpected' )->inContentLanguage()->text()
+			);
+		}
+
+		if ( !$status ) {
+			$this->logFailToStaff();
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Checks if the request provided to the constructor is valid.
+	 *
+	 * @return bool True if all prerequisites are met
+	 */
+	protected function setup() {
+		$this->setExecutionLimits();
+
+		$oldUserName = $this->getOldUserName();
+		$newUserName = $this->getNewUserName();
+
+		if ( !$this->areUserNamesValid( $oldUserName, $newUserName ) ) {
+			return false;
+		}
+
+		// validate new username and disable validation for old username
+		$oldUser = \User::newFromName( $oldUserName, false );
+		$newUser = \User::newFromName( $newUserName, 'creatable' );
+
+		if ( !$this->areUserObjectsValid( $oldUser, $newUser ) ) {
+			return false;
+		}
+
+		$uid = $this->getOldUserId( $oldUserName, $oldUser );
+
+		if ( $uid == 0 ) {
+			$this->addUserDoesNotExistError();
+			return false;
+		};
+
+		if ( !$this->checkOldUserPermissions( $oldUser ) ) {
+			return false;
+		}
+
+		$this->setNamesAndIds( $uid, $newUser, $oldUser );
+
+		// If there are only warnings and user confirmed that, do not show them again on the success page ;-)
+		if ( $this->actionConfirmed ) {
+			$this->warnings = [];
+		} elseif ( count( $this->warnings ) ) {
+			// In case the action is not confirmed and there are warnings, display them and wait
+			// for confirmation before running the process
+			return false;
+		}
+
+		return empty( $this->errors );
+	}
+
+	/**
+	 * Make sure the process will not be stopped in the middle of renaming the user
+	 */
+	private function setExecutionLimits() {
+		set_time_limit( self::MAX_EXECUTION_TIME );
+		ignore_user_abort( true );
+		ini_set( "max_execution_time", self::MAX_EXECUTION_TIME );
+	}
+
+	/**
+	 * Get the old username, sanitizing the username input
+	 *
+	 * @return String
+	 */
+	private function getOldUserName() {
+		// Sanitize input data
+		$oldTitle = \Title::makeTitle( NS_USER, trim( str_replace( '_', ' ', $this->requestData->oldUsername ) ) );
+		$oldUserName = is_object( $oldTitle ) ? $oldTitle->getText() : '';
+		$this->phalanxTest( $oldUserName );
+
+		return $oldUserName;
+	}
+
+	/**
+	 * Get the new username, sanitizing the username input
+	 *
+	 * @return String
+	 */
+	private function getNewUserName() {
+		global $wgContLang;
+
+		// Force uppercase of new username, otherwise wikis with wgCapitalLinks=false can create lc usernames
+		$newTitle = \Title::makeTitleSafe( NS_USER, $wgContLang->ucfirst( $this->requestData->newUsername ) );
+		$newUserName = is_object( $newTitle ) ? $newTitle->getText() : '';
+		$this->antiSpoofTest( $newUserName );
+		$this->phalanxTest( $newUserName );
+
+		return $newUserName;
+	}
+
+	private function areUserNamesValid( $oldUserName, $newUserName ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)